ENIGMAsystem/SHELL/Platforms/Cocoa/CocoaJoystick.cpp

```diff
@@ -0,0 +1,72 @@
+/** Copyright (C) 2016 Rodrigo Rocha
+***
+*** This file is a part of the ENIGMA Development Environment.
+***
+*** ENIGMA is free software: you can redistribute it and/or modify it under the
+*** terms of the GNU General Public License as published by the Free Software
+*** Foundation, version 3 of the license or any later version.
+***
+*** This application and its source code is distributed AS-IS, WITHOUT ANY
+*** WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+*** FOR A PARTICULAR PURPOSE. See the GNU General Public License for more
+*** details.
+***
+*** You should have received a copy of the GNU General Public License along
+*** with this code. If not, see <http://www.gnu.org/licenses/>
+**/
+
+#include <string>
+using namespace std;
+
+namespace enigma
+{
+  //stub
+}
+
+/* Note: empty stub functions to build Mac executables without errors (and no joystick support, of course) */
+namespace enigma_user
+{
+  bool joystick_load(int id) {
+    return false; //stub
```

Comment:
Well, the game was building on windows/linux but not mac because of the lack of joystick support. Unless the user remove manually all the joystick code from the game I think it is necessary while there is no platform parity...  Write once, compile anywhere :)

By the way, I have some experience with Game Controllers on Mac, I can do that in the future.